src/Core/Name.idr

```diff
@@ -78,6 +112,7 @@ export
 userNameRoot : Name -> Maybe UserName
 userNameRoot (NS _ n) = userNameRoot n
 userNameRoot (UN n) = Just n
+userNameRoot (OP _ n) = Just (Basic n) -- operator and fixities are user names
```

Comment:
I am really uneasy about this. This mean that everywhere we used to match
on `UN` we should now match on both `UN` and `OP`. How do we know we
are not missing cases that are now handled by catchalls?

Reply:
Currently only fixity declarations use the `OP` constructor. I started porting the rest of the parser to use `OP` as well (also because it was mentionned in TODO in comments) but that was too much work, so right now all the machinery uses `UN` as before and `OP` never appears.

This suggests that `OP` should probably be `Fixity -> String -> Name` rather than `Maybe fixity`.

Reply:
Ah yes, this version bypasses that in two ways:
- when declaring fixities you are not allowed to use namespaced backticked names
- When looking up fixity conflicts we always use the nameRoot so it does not matter if the operator used is a namespaced backticked name or a plain operator

Here is an example where this is handled: https://github.com/idris-lang/Idris2/pull/2997/files#diff-6a3fe63c93fc56d8ac342e2910c5e4957fbcda5a72480912648a24b499b79880R317

Reply:
I've removed `OP` and I'm now relying on querying the namespace. I'm not sure this is ideal either, this is probably worth thinking about